content/documentation/pages/3-stream-developer-guides/2-streams/2-standalone-stream-kafka.md

```diff
@@ -278,33 +353,175 @@ Now let's build the Usage Cost Logger application. In the directory `usage-cost-
+    		ArgumentCaptor<UsageCostDetail> captor = ArgumentCaptor.forClass(UsageCostDetail.class);
+    		this.sink.input().send(MessageBuilder.withPayload("{\"userId\":\"user3\",\"callCost\":10.100000000000001,\"dataCost\":25.1}").build());
+    		verify(this.usageCostLogger).process(captor.capture());
+    	}
+
+    	@EnableAutoConfiguration
+    	@EnableBinding(Sink.class)
+    	static class TestConfig {
+
+    		// Override `UsageCostLogger` bean for spying.
+    		@Bean
+    		@Primary
+    		public UsageCostLogger usageCostLogger() {
+    			return spy(new UsageCostLogger());
+    		}
+    	}
+    }
+
+    ```
+
+1.  The test case `contextLoads` verifies the application starts successfully.
+1.  The test case `testUsageCostLogger` verifies if the `process` method of `UsageCostLogger` is invoked using `Mockito`.
+    To do this, the static class `TestConfig` overrides the existing `UsageCostLogger` bean to create a Mock bean of `UsageCostLogger`.
+    Since we are mocking the `UsageCostLogger` bean, the `TestConfig` also annotates `@EnableBinding` and `@EnableAutoConfiguration` explicitly.
+
+## Deployment
+
+In this section we will deploy the apps created above to the local machine, Cloud Foundry and Kubernetes.
+
+When you deploy these three applications (UsageDetailSender, UsageCostProcessor and UsageCostLogger), the flow of message looks like this:
 
 ```
-spring.cloud.stream.bindings.input.destination=test-usage-cost
+UsageDetailSender -> UsageCostProcessor -> UsageCostLogger
 ```
 
-#### Running the Sink
+The source application `UsageDetailSender`'s output is connected to the `UsageCostProcessor` processor application's input.
+The `UsageCostProcessor` application's output is connected to the `UsageCostLogger` sink application's input.
 
-You can run the standalone `UsageCostLogger` sink application as,
+Since we use Kafka as the streaming platform, we have selected Spring Cloud Stream `Kafka` binder for this application.
+When these applications are run, the `Kafka` binder binds the applications output/input boundaries into the corresponding destinations (topics) at Kafka.
+
+Let's choose our destinations at Kafka:
+
+- UsageDetailSender's output and `UsageCostProcessor`'s input are bound to `test-usage-detail` topic
+- UsageCostProcessor's output and `UsageCostLogger`'s input are bound to `test-usage-cost` topic.
+
+### Local
+
+You can run the above applications as standalone applications on your `local` environment.
+
+You can [download](https://kafka.apache.org/downloads) and setup `Kafka` on your local.
+
+After unpacking the downloaded archive, you can start `ZooKeeper` and `Kafka` servers as follows:
 
 ```
-java -jar target/usage-cost-logger-kafka-0.0.1-SNAPSHOT.jar --spring.cloud.stream.bindings.input.destination=test-usage-cost &
+./bin/zookeeper-server-start.sh config/zookeeper.properties &
 ```
 
-Now, you can see that this application logs the usage cost detail.
+```
+./bin/kafka-server-start.sh config/server.properties &
```

Comment:
I'm still not a fan of background processes. It shouldn't be hidden from the users who are trying it - better to see all the console output.